march_hardware/include/march_hardware/PDOmap.h

```diff
@@ -16,13 +16,15 @@ namespace march
 struct IMCObject
 {
   uint16_t address;           // in IMC memory (see IMC manual)
-  uint16_t length;            // bits (see IMC manual)
+  uint8_t sub_index;          // sub index corresponding to PDO register (see IMC manual)
+  uint8_t length;             // bits (see IMC manual)
   uint32_t combined_address;  // combine the address(hex), sub-index(hex) and length(hex)
 
-  explicit IMCObject(uint16_t _address, uint16_t _length) : address(_address), length(_length)
+  explicit IMCObject(uint16_t _address, uint8_t _sub_index, uint8_t _length)
+    : address(_address), sub_index(_sub_index), length(_length)
```

Comment:
Now that we are changing this anyway, it would be nice to remove the [`explicit`](https://en.cppreference.com/w/cpp/language/explicit), since that is only meaningful on constructors with one paramater.
```suggestion
  IMCObject(uint16_t _address, uint8_t _sub_index, uint8_t _length)
    : address(_address), sub_index(_sub_index), length(_length)
```